src/ceph-detect-init/ceph_detect_init/__init__.py

```diff
@@ -123,10 +134,23 @@ def platform_information():
         return ('docker', 'docker', 'docker')
 
     if platform.system() == 'Linux':
-        linux_distro = platform.linux_distribution(
-            supported_dists=platform._supported_dists + ('alpine', 'arch'))
-        logging.debug('platform_information: linux_distribution = ' +
-                      str(linux_distro))
+        linux_distro = ('', '', '')
+        if os.path.isfile('/etc/os-release'):
+            try:
+                with open('/etc/os-release', 'r') as f:
+                    data = f.read()
+                linux_distro = (
+                    _extract_from_os_release(data, 'ID'),
+                    _extract_from_os_release(data, 'VERSION'),
```

Comment:
It's possible for `VERSION` to not be present (it's not there in /etc/os-release on openSUSE Tumbleweed), which probably means the SUSE version of `choose_init()` will do the wrong thing. Maybe try using `VERSION_ID` instead. There's a bunch of example os-release files in https://github.com/ceph/ceph-deploy/pull/453/files if it helps. But otherwise, I'm happy with just parsing os-release rather than using distro from pypi for the reasons you've stated.